### DISCUSSION TITLE
Functions doesn't work. NO access

### DISCUSSION THREAD

**Discussion by SavixDR** [supabase/supabase#30269] [Questions]
I have a function called 'view_products' I created that function in supabase. I want to call that function using supabase.rpc but for authorization error it returns 403. The problem is I have done this request as a anon user earlier. In here I have disabled all the RLS for every table because I'm still on the development stage. I have also grant access for the function for anon users too. But still I cannot get access. Still shows 403 error code.  I have tried after creating a login as well as a authenticated user still this error shows.

**Comment by GaryAustin1** [comment]
Can you call the function from the SQL editor?
Can you call it with service_role?
Did you change the grants on the function?
If you recreate the function does it work?

Also can you provide the detail error message you are getting for that 403.
